### PR TITLE
Admin sales fees

### LINKED ISSUE #4884: Admin: Sales: Implement fees route
**Is your feature request related to a problem? Please describe.**
The admin page has no working fees tab as the route is missing

### PR DESCRIPTION
**Pull request by maxlorenz**
Fixes #4884 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
The admin sales had no fees route that has revenue and fees percentage information for every event

#### Changes proposed in this pull request:

- Move revenue calculation to better class
- Add revenue to order model
- Add new route for fees for the admin sales

### CODE REVIEW

File: app/models/event.py [hunk 1]
```diff
@@ -67,10 +66,12 @@ class Event(db.Model):
...
-    event_type_id = db.Column(db.Integer, db.ForeignKey('event_types.id', ondelete='CASCADE'))
-    event_topic_id = db.Column(db.Integer, db.ForeignKey('event_topics.id', ondelete='CASCADE'))
-    event_sub_topic_id = db.Column(db.Integer, db.ForeignKey(
-        'event_sub_topics.id', ondelete='CASCADE'))
+    event_type_id = db.Column(
+        db.Integer, db.ForeignKey('event_types.id', ondelete='CASCADE'))
+    event_topic_id = db.Column(
+        db.Integer, db.ForeignKey('event_topics.id', ondelete='CASCADE'))
+    event_sub_topic_id = db.Column(
+        db.Integer, db.ForeignKey('event_sub_topics.id', ondelete='CASCADE'))
```
Comment (srv-twry) [inline app/models/event.py]: I think we should avoid making unrelated changes specially formatting ones in the PR. This not only makes reviewing complicated but also destroys git blame, in other words puts the blame on you. :wink:

File: app/models/event.py [hunk 2]
```diff
@@ -260,10 +270,20 @@ def __str__(self):
...
             super(Event, self).__setattr__(name, value)
 
+    @property
+    def fee(self):
+        """
+        Returns the fee as a float from 0 to 1 for this event
+
+        Is retrieved from the db using the `payment_currency`
+        """
+        return get_fee(self.payment_currency)
```
Comment (srv-twry) [inline app/models/event.py]: So you're basically returning the service_fee. Why is it always between 0 to 1 ? Is it a fraction of the ticket fee that is provided to eventyay ? 
If no, then please explain. If yes, then I think the docs are extremely misleading. https://open-event-api-dev.herokuapp.com/#ticket-fees
```
"attributes": {
        "currency": "USD",
        "service-fee": 3.14,
        "maximum-fee": 5.14
      },
```

The sample response has values greater than 1 :(

Comment (maxlorenz) [reply app/models/event.py]: ah my mistake, just saw that. will fix it!

**Comment by Kreijstal** [comment]
This now has conflicts..

File: app/api/admin_sales/fees.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        inflect = dasherize
+
+    id = fields.String()
+    name = fields.String()
+    payment_currency = fields.String()
+    fee = fields.Method('format_fee')
+    revenue = fields.Method('calc_revenue')
+    ticket_count = fields.Method('calc_ticket_count')
+
+    @staticmethod
```
Comment (poush) [inline app/api/admin_sales/fees.py]: Is it require for them to be staticmethod ?

File: app/api/helpers/scheduled_jobs.py [hunk 4]
```diff
@@ -2,14 +2,14 @@
 from app.models.order import Order
 from app.models.event_invoice import EventInvoice
 from app.models.ticket import Ticket
+from app.models.ticket_fee import get_fee
```
Comment (poush) [inline app/api/helpers/scheduled_jobs.py]: What's the need to import it here?

Comment (maxlorenz) [reply app/api/helpers/scheduled_jobs.py]: `get_fee` is used as a scheduled job to send out the fees notification so this shares the same calculation

File: app/api/admin_sales/fees.py (hunk 3 above)
Comment (maxlorenz) [reply app/api/admin_sales/fees.py]: It could be a `@property` as well

Comment (maxlorenz) [reply app/api/admin_sales/fees.py]: Yes, as they don't use the class itself (`self` would not be used)

File: app/models/event.py (hunk 2 above)
Comment (maxlorenz) [reply app/models/event.py]: fixed. can this be merged now?
